src/views/home/index.js

```diff
@@ -3,37 +3,54 @@ import React, { useEffect, useState, useContext } from 'react'
 import { AuthContext } from '../../providers/use-auth'
 import { Navigate } from "react-router-dom";
 import './home.css'
-import { getDog } from '../../utils/fetch-request';
+import { useInitDog } from '../../hooks';
+import { usePreloadDog } from '../../hooks';
 
 export const Home = () => {
   const { user } = useContext(AuthContext)
   const [left, setLeft] = useState(null)
   const [mid, setMid] = useState(null)
   const [right, setRight] = useState(null)
+  const dogs = useInitDog()
+  const [preloadDogs, setPreloadDogs] = usePreloadDog()
+  const preLoadImg = new Image()
 
   useEffect(() => {
-    Promise.all([getDog(), getDog(), getDog()])
-      .then(dogs => {
-        const [leftDog, midDog, rightDog] = dogs.map(dog => dog.message)
-  
-        setLeft(leftDog)
-        setMid(midDog)
-        setRight(rightDog)
-      })
-  }, [])
+    if (!dogs.length) return
+
+    setLeft(dogs[0])
+    setMid(dogs[1])
+    setRight(dogs[2])
+  }, [dogs])
+
+  useEffect(() => {
```

Comment:
@wood-crafter you can refactor the code like this:

```jsx
useEffect(() => {
  // preloadDogs won't do anything if it's empty
  // So we can even skip the length check
  preloadDogs.forEach(url => {
    preLoadImg.src = url
  })
}, [preloadDogs, preLoadImg])
```

Don't worry I'm working on this already...!

Reply:
> @wood-crafter you can refactor the code like this:
> 
> ```js
> useEffect(() => {
>   // preloadDogs won't do anything if it's empty
>   // So we can even skip the length check
>   preloadDogs.forEach(url => {
>     preLoadImg.src = url
>   })
> }, [preloadDogs, preLoadImg])
> ```
> 
> Don't worry I'm working on this already...!

I usually worry about main problem, and forgeting other - lead to problem like this :( Thank you anyway